bin/post_process_sce.R

```diff
@@ -96,20 +96,43 @@ metadata(sce)$min_gene_cutoff <- opt$gene_cutoff
 # create adt_scpca_filter column, if CITEseq ----------
 alt_exp <- opt$adt_name
 if (alt_exp %in% altExpNames(sce)) {
-  sce$adt_scpca_filter <-  ifelse(
-    altExp(sce, alt_exp)$discard,
-    "Remove",
-    "Keep"
-  )
   
-  # assign `adt_scpca_filter_method` metadata based on colData contents
-  if ("sum.controls" %in% names(colData(altExp(sce, alt_exp)))) {
-    metadata(sce)$adt_scpca_filter_method <- "cleanTagCounts with isotype controls"
-  } else if ("ambient.scale" %in% names(colData(altExp(sce, alt_exp)))) {
-    metadata(sce)$adt_scpca_filter_method <- "cleanTagCounts without isotype controls"
+  # check for NAs in filtering columns
+  use_discard <- sum(is.na(altExp(sce, alt_exp)$discard)) == 0
+  use_zero.ambient <- sum(is.na(altExp(sce, alt_exp)$zero.ambient)) == 0
+  
+  # Fail right away if both are FALSE
+  if (!use_discard & !use_zero.ambient) {
+    warning("Failed to filter ADT data.")
+    sce$adt_scpca_filter <- NULL
```

Comment:
I think it will cause fewer problems downstream to leave this column present, with the warning and `"No filter"`  for the method sufficient.
```suggestion
    sce$adt_scpca_filter <- "Keep"
```

Reply:
Done in a93152c (not via suggestion).